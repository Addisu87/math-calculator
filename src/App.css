:root {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

* {
  --dark: rgb(235, 229, 229);
}

.App {
  width: 100%;
  height: 100vh;
  background-color: var(--dark);
  display: flex;
  align-items: center;
  justify-content: center;
}
