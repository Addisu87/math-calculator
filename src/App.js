import { Route, Routes } from 'react-router-dom';
import './App.css';
import Navbar from './components/Navbar';
import Home from './components/Home';
import Calculator from './components/Calculator';
import Quotes from './components/Quotes';

function App() {
  return (
    <div className="App">
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="calculator" element={<Calculator />} />
        <Route path="quotes" element={<Quotes />} />
      </Routes>
    </div>
  );
}

export default App;
