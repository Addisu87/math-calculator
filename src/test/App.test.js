import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';

describe('Body of the whole App', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <App />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Expect it to render the home page', async () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );
    expect(
      screen.getByText(
        /Math magicians is a website for all fans of mathematics./,
      ),
    ).not.toBeNull();
  });

  it('Making sure the quote page renders', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    const quote = screen.getByText('Quotes');

    fireEvent.click(quote);

    const speaker = screen.getByText(/Albert Einstein/);
    expect(speaker).not.toBeNull();
  });
});
