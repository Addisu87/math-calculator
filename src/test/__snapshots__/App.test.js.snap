// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="sc-bczRLJ gUDtTg"
  >
    <h2
      className="sc-gsnTZi dayyfU"
    >
      Math Magicians
    </h2>
    <div
      className="sc-dkzDqf bxKzAV"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home |
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator |
      </a>
      <a
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </div>
  </div>
  <div
    className="sc-hKMtZM buVsdN"
  >
    <h4
      className="sc-eCYdqJ jBlsKL"
    >
      Welcome to our page!
    </h4>
    <article>
      <p>
        Emmet Abbreviation Lorem, ipsum dolor sit amet consectetur adipisicing elit. Praesentium facere numquam nobis minus ipsam quaerat doloremque aliquam suscipit vero, vitae dolorem ab laboriosam laudantium voluptatum magni libero iusto error eos?
      </p>
      <br />
      <p>
        Emmet Abbreviation Lorem, ipsum dolor sit amet consectetur adipisicing elit. Praesentium facere numquam nobis minus ipsam quaerat doloremque aliquam suscipit vero, vitae dolorem ab laboriosam laudantium voluptatum magni libero iusto error eos?
      </p>
    </article>
  </div>
</div>
`;
